internal/paascharm/gogen_test.go

```diff
@@ -0,0 +1,33 @@
+package paascharm_test
+
+import (
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/canonical/paascharmgen/internal/paascharm"
+)
+
+func TestCreateGoStructsWithEmptyCharmcraftYaml(t *testing.T) {
+	tmpdir := t.TempDir()
+	charmcraftFile := path.Join(tmpdir, "charmcraft.yaml")
+	os.WriteFile(charmcraftFile, []byte(""), 0644)
```

Comment:
You should probably handle the error from `WriteFile`.

Reply:
ops, totally missed that one. Thanks